java-cfenv-boot/src/main/java/io/pivotal/cfenv/spring/boot/RedisCfEnvProcessor.java

```diff
@@ -29,12 +29,82 @@
  * @author Mark Pollack
  * @author Scott Frederick
  */
-public class RedisCfEnvProcessor implements CfEnvProcessor {
+public class RedisCfEnvProcessor {
 
-	private static String[] redisSchemes = { "redis", "rediss" };
+	private final static String[] redisSchemes = { "redis", "rediss" };
 
-	@Override
-	public boolean accept(CfService service) {
+	public final static class Boot2 implements CfEnvProcessor {
+
+		private static final int BOOT_VERSION = 2;
+		private static final String PREFIX = "spring.redis";
+
+		@Override
+		public boolean accept(CfService service) {
+			if (!SpringBootVersionResolver.isBootMajorVersionEnabled(BOOT_VERSION)) {
+				return false;
+			}
+			return RedisCfEnvProcessor.accept(service);
+		}
+
+		@Override
+		public void process(CfCredentials cfCredentials, Map<String, Object> properties) {
+
+			RedisCfEnvProcessor.process(cfCredentials, properties, PREFIX);
+
+			// TODO: what do we do with those? drop them ?
+			//			map.from("client-name").to("spring.redis.client-name");
+			//			map.from("cluster.max-redirects").to("spring.redis.cluster.max-redirects");
+			//			map.from("cluster.nodes").to("spring.redis.cluster.nodes");
+			//			map.from("database").to("spring.redis.database");
+			//			map.from("sentinel.master").to("spring.redis.sentinel.master");
+			//			map.from("sentinel.nodes").to("spring.redis.sentinel.nodes");
+			//			map.from("url").to("spring.redis.url");
+		}
+
+		@Override
+		public CfEnvProcessorProperties getProperties() {
+			return RedisCfEnvProcessor.getProperties(PREFIX);
+		}
+	}
+
+	/**
+	 * This is a special case for Boot 3.
+	 */
+	public final static class Boot3 extends SpringBootVersionResolver implements CfEnvProcessor {
+
+		private static final int BOOT_VERSION = 3;
+		private static final String PREFIX = "spring.data.redis";
+
+		@Override
+		public boolean accept(CfService service) {
+			if (!SpringBootVersionResolver.isBootMajorVersionEnabled(BOOT_VERSION)) {
+				return false;
+			}
+			return RedisCfEnvProcessor.accept(service);
+		}
+
+		@Override
+		public void process(CfCredentials cfCredentials, Map<String, Object> properties) {
+
+			RedisCfEnvProcessor.process(cfCredentials, properties, PREFIX);
+
+			// TODO: what do we do with those? drop them ?
+			//				map.from("client-name").to("spring.data.redis.client-name");
+			//				map.from("cluster.max-redirects").to("spring.data.redis.cluster.max-redirects");
+			//				map.from("cluster.nodes").to("spring.data.redis.cluster.nodes");
+			//				map.from("database").to("spring.data.redis.database");
+			//				map.from("sentinel.master").to("spring.data.redis.sentinel.master");
+			//				map.from("sentinel.nodes").to("spring.data.redis.sentinel.nodes");
+			//				map.from("url").to("spring.data.redis.url");
```

Comment:
The best place to start thinking about this is to check out what the [Redis Tile on TAS](https://docs.vmware.com/en/Redis-for-VMware-Tanzu-Application-Service/index.html) actually provides, and see if we need those extra props.

Note: Redis, the company, also packages [a Tile](https://docs.pivotal.io/partners/redis-labs-enterprise-pack/index.html), maybe take a look there as well.